otel-extension/src/main/scala/io/scalac/mesmer/agent/util/i13n/OtelTypeInstrumentation.scala

```diff
@@ -0,0 +1,46 @@
+package io.scalac.mesmer.agent.util.i13n
+
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer
+import net.bytebuddy.description.`type`.TypeDescription
+import net.bytebuddy.matcher.ElementMatcher
+
+import io.scalac.mesmer.core.typeclasses.Encode
+
+case class OtelTypeInstrumentation private (
+  private[i13n] val instrumentedType: TypeDesc,
+  private[i13n] val adviceSet: Set[AdviceApplication]
+) {
+  def using(advice: AdviceApplication): OtelTypeInstrumentation =
```

Comment:
I like the fact that we have a separate type. It's helpful if you want to (eg.) extract a value for the advice and reuse it (see the tests).

Reply:
I don't suggest to remove `AdviceApplication` altogether, but to, but rather to make it private. In the user code we'll end up then with:
```
OtelTypeInstrumentation
.instrument(named("SomeType")
.using(named("method1Name"), "path.to.advice.One")
.using(named("method2Name"), "path.to.advice.Two")
.encode()

```,
instead of:
```
OtelTypeInstrumentation
.instrument(named("SomeType")
.using(AdviceApplication .advice(named("method1Name"), "path.to.advice.One"))
.using(AdviceApplication named("method2Name"), "path.to.advice.Two"))
.encode()
``` 

with all the underlying structure intact - just a little less boilerplate 🙃

Reply:
We loose the ability to pass advice "description" as an reusable argument. So in testing even if we have this: 

```
private val adviceExample = advice(ElementMatchers.named("fooFunction"), "AdviceClass")
```

we now always have to do this: 

```
.using(adviceExample.method, adviceExample.adviceClassName)
```

IMO that's less comfortable to use. To me advice is a first-class citizen here - no need to hide it - we can embrace it's existence as I did in testing x)

What would you say for this approach: b384a3db ? Imho it's even more concise and lets us reuse code more.